### PR TITLE
doc: Convert samples/ to CMake

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

File: samples/boards/arduino_101/environmental_sensing/README.rst [hunk 1]
```diff
@@ -106,17 +106,15 @@ Before using the environmental sensing sample, the Arduino 101 board needs to be
...
-or using DFU. (DFU is preferred).
-
-The sample can be found here: :file:`samples/boards/arduino_101/environmental_sensing` in the
-Zephyr project tree.
-
-.. code-block:: console
-
-   $ make
-   $ make flash # for flashing with JTAG
+The sample provide a single :file:`CMakeLists.txt` that builds both images,
```
Comment (carlescufi) [inline samples/boards/arduino_101/environmental_sensing/README.rst]: @nashif @SebastianBoe is this correct?

Comment (nashif) [reply samples/boards/arduino_101/environmental_sensing/README.rst]: no

Comment (carlescufi) [reply samples/boards/arduino_101/environmental_sensing/README.rst]: Fix doc or CMakeLists then?

**Comment by carlescufi** [comment]
@nashif Fixed it by explicitly building both cores separately

**Review by dbkinder — APPROVED** [review]

File: samples/boards/arduino_101/environmental_sensing/README.rst (hunk 1 above)
Comment (SebastianBoe) [reply samples/boards/arduino_101/environmental_sensing/README.rst]: I suggest we fix doc. And then use this as an example of multi-isa builds when we support this in the next release.

Docs can say something like:

Build and flash both the "sensor" application and the "ap" application in sequence.

Comment (SebastianBoe) [reply samples/boards/arduino_101/environmental_sensing/README.rst]: Ah, didn't see this was resolved :)

File: samples/basic/threads/README.rst [hunk 2]
```diff
@@ -22,7 +22,8 @@ Each thread is then defined at compile time using K_THREAD_DEFINE.
 Building
 ********
 
-.. code-block:: console
-
-    $ cd samples/basic/threads
-    $ make
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/threads
+   :board: 96b_carbon
```
Comment (mbolivar) [inline samples/basic/threads/README.rst]: Might want to add a follow-up patch that makes clear it's an example for this board, the same as done elsewhere in this PR.

**Review by mbolivar — COMMENTED** [review]
Looks great; nicely done! Just one minor comment I would have pointed out, that I wouldn't have wanted to block the merge.

**Pull request merged**